future-apps/payroll/contracts/Payroll.sol

```diff
@@ -594,21 +595,54 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
         emit TerminateEmployee(_employeeId, employee.accountAddress, _endDate);
     }
 
-    function _getOwedSalary(uint256 _employeeId, uint64 _date) internal view returns (uint256) {
+    /**
+     * @dev Calculates the date timestamp corresponding to the requested paying amount based on the employee's last payroll date.
+     * @param _employeeId Employee's identifier.
+     * @param _payedAmount Amount payed to the employee to query the timestamp of.
+     * @return The date timestamp in seconds corresponding to the given paying amount based on the employee's last payroll date.
+     */
+    function _getLastPayroll(uint256 _employeeId, uint256 _payedAmount) internal view returns (uint64) {
+        Employee storage employee = employees[_employeeId];
+
+        uint256 timeDiff = _payedAmount.div(employee.denominationTokenSalary);
```

Comment:
I don't get it: doesn't Solidity always truncate?
Anyway, I agree this doesn't look good. I think we should try to avoid relying on division.

Reply:
It's exactly the truncating that causes this behaviour.

If you had a salary per second of 100 USD, I could request 99 USD here to get a `timeDiff` of 0.

Reply:
> It's exactly the truncating that causes this behaviour.

Yes, that's why I didn't understand this:

> We should check if the division was perfect, and if not, floor it.

Were you proposing to truncate that division? (As numbers are always positive I understand flooring and truncating are synonyms in our context)

Reply:
Oops, amended to use "ceiling" instead, floor was the wrong direction 😄

Reply:
Oh, ok, then it makes sense. Although it would be still inaccurate. I would keep the flooring and then adjust the real amount payed to that `timeDiff`, which in that example it would be zero and would always be less than requested.
Btw, if I had a salary of $100 per second, trust me, I wouldn't bother doing this trick, hahaha. This is of course a joke, but it's worth noting that the maximum error will be small, at most the amount of salary per second, which except for sport stars and other big salaries  would be less than a penny.
Another solution, easy but still slightly inaccurate, would be to define a minimum periodicity (i.e., `timeDiff`) for requesting salary. (I still like better the previous one)

Reply:
> The maximum error will be small, at most the amount of salary per second

Regardless of how small this is, you could do this infinitely many times as gas allowed.

If we go with our rough estimate that you can make 20 calls per block, you could fill entire blocks for as long as you could to "siphon" out money from the organization.

Reply:
Yes, yes, I know, I never meant to leave it as it is (although I guess most of the times the cost of gas would be higher).

Reply:
Moving this to a separate issue